pkgs/by-name/tr/treecat/package.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, fetchFromSourcehut
+, hare
+, harec
+, lib
+, scdoc
+}:
+let
+  # treecat needs a more recent version of Hare. Until #265124 is merged onto
+  # master (it's already merged on staging), these overrides are needed.
+  harec' = harec.overrideAttrs (_old: {
+    version = "unstable-2023-11-28";
+    src = fetchFromSourcehut {
+      owner = "~sircmpwn";
+      repo = "harec";
+      rev = "ec3193e3870436180b0f3df82b769adc57a1c099";
+      hash = "sha256-HXQIgFC4YVDJjo5xbyg1ea3jWYKLEwKkD1KFzWFz9UI=";
+    };
+  });
+  hare' = (hare.overrideAttrs (_old: {
+    version = "unstable-2023-11-27";
+    src = fetchFromSourcehut {
+      owner = "~sircmpwn";
+      repo = "hare";
+      rev = "d94f355481a320fb2aec13ef62cb3bfe2416f5e4";
+      hash = "sha256-Mpl3VO4xvLCKHeYr/FPuS6jl8CkyeqDz18mQ6Zv05oc=";
+    };
+    # checkPhase needs the tzdata file added on the aforementioned PR.
+    doCheck = false;
+  })).override { harec = harec'; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "treecat";
+  version = "1.0.2-unstable-2023-11-28";
+
+  outputs = [ "out" "man" ];
+
+  src = fetchFromSourcehut {
+    owner = "~autumnull";
+    repo = "treecat";
+    rev = "d277aed99eb48eef891b76916a61029989c41d2d";
+    hash = "sha256-4A01MAGkBSSzkyRw4omNbLoX8z+pHfoUO7/6QvEUu70=";
+  };
+
+  nativeBuildInputs = [
+    hare'
+    scdoc
+  ];
+
+  preBuild = ''
+    HARECACHE="$(mktemp -d)"
+    export HARECACHE
+    export PREFIX="${builtins.placeholder "out"}"
+  '';
+
+  # Until #269767 is merged, we cannot make use of treecat's build framework: haredo.
```

Comment:
should be merged soon :)

Reply:
Looking forward to it!

Reply:
Added `haredo` to `nativeBuildInputs`: now there's no need for overwriting the build, check and install phases. Thanks for the merge!